### ISSUE TITLE
Isochrone for a destination point instead of origin. 

### ISSUE THREAD

**Issue by ayl** [valhalla/valhalla#3475]
It seems that the current implementation of isochrones is for an origin point but there are some business and research related use-cases where the isochrone for a destination point is more important. (eg what is the 15 minute catchment area for a restaurant?) 

It may be a useful feature to add in the future for the isochrone API.

**Comment by kevinkreiser** [comment]
The code has both support for origins and destinations but we just havent exposed a parameter for you to select the destination mode in the external (http, python, c++ actor) apis. we can accomplish this with a simple additional parameter, we should probably also make this happen if you select "arrive_by" `date_time` `type`. The code change should mainly happen here: https://github.com/valhalla/valhalla/blob/master/src/thor/isochrone_action.cc#L36-L37 in addition to the request parameter parsing.

**Comment by Yosef-Irawan** [comment]
Hi @kevinkreiser, I have tested the reverse isochrone. but I got a strange result. the result is much smaller than the result that is not reversed. I have tested the reverse isodistance also.  I got a nice result from reverse isodistance. 
![Screenshot from 2022-02-03 16-46-01](https://user-images.githubusercontent.com/73457493/152318861-ea3332ed-f146-41da-b0d2-4f1dbb63ceae.png)
Legend:
- gray : isodistance 10 km
- purple : reverse isodistance 10 km
- red : isochrone 5 min
- yellow : reverse isochrone 5 min

do you have any advice on how I should fix this issue?

**Comment by nilsnolde** [comment]
Thanks, would be cool to have this done! Can you pls put a PR with the code changes? Then we can take a closer look. 

Some differences are of course expected due to direction-dependent access and oneways, but the isochrones look a little too much 😅

**Comment by Yosef-Irawan** [comment]
https://github.com/valhalla/valhalla/pull/3528/files @nilsnolde  this is the PR, sorry I haven't added any tests code for it.

**Comment by Yosef-Irawan** [comment]
Hi @nilsnolde @kevinkreiser, I found a strange thing when tracing in the reverse transition cost calculation. I found some strange cost that cost has 0 in cost but have 27 sec in time. that is not happening in forward mode. after I take a look deeper, the time calculation takes u turn. but cost calculation takes 0 because the stop impact is 0. Do you have any suggestions to solve this issue?

**Comment by kevinkreiser** [comment]
@Yosef-Irawan can you share the test case or the location in the graph where it occurs?

**Comment by Yosef-Irawan** [comment]
@kevinkreiser, this is the test case
```
curl -L -X POST 'http://localhost:8002/isochrone' -H 'Content-Type: application/json' --data-raw '{
  "locations": [
    {
      "lat":-6.889233,
      "lon": 107.596837

    }
  ],
  "costing": "auto",
  "contours":[{"time":5}],
  "costing_options": {
    "auto": {
      "country_crossing_penalty": 2000.0
    }
  },
  "units": "kilometers",
  "polygons": true,
  "id": "my_work_route",
  "isochrone_type": "reverse"
}'
```

**Comment by Yosef-Irawan** [comment]
![Screenshot from 2022-02-21 20-50-26](https://user-images.githubusercontent.com/73457493/154968319-e29b5a4e-18b5-408e-843a-184e3bfc8509.png)
this is the strange transition cost calculation. I think it is the u-turn transition but why the stop impact is zero?

**Comment by Yosef-Irawan** [comment]
Hi @kevinkreiser @nilsnolde, I have found the root cause for the issue. I have submitted the changes. Please review this 
https://github.com/valhalla/valhalla/pull/3528/commits/2b0f9b3baaab0b7eff444cf580a2288e390dc832. :bow:

**Comment by nilsnolde** [comment]
this was merged with #3528
